code/graphics/render.cpp

```diff
@@ -587,6 +588,27 @@ static void gr_string_old(float sx,
 	gr_set_2d_matrix();
 
 	float scale_factor = (canScale && !Fred_running) ? get_font_scale_factor() : 1.0f;
+
+	if (canAutoScale && !Fred_running) {
+		// Lambda to calculate font size based on screen resolution
+		auto calculateAutoSize = [height]() -> float {
+			int vmin = std::min(gr_screen.max_w, gr_screen.max_h);
+
+			// Base size calculation (similar to ~Npx font at 1080p)
+			// Use 1080p because that's generally what font sizes have been targeting for years
+			// And should provide a fairly close out-of-the-box solution
+			float baseSize = vmin * (height / 1080.0f);
+			return std::round(baseSize);
+		};
+
+		// Calculate the auto scale factor
+		float autoSizedFont = calculateAutoSize();
```

Comment:
Err, why is this a lambda?
If you create the lambda and immediately execute it without storing it anywhere, why not just run the code directly?

Reply:
Ah cause I just copied it from where I used it last time without thinking about it much. 

Honestly I should make both places use the same code so if we ever need to change the formula we only have to do it once